sdk/identity/Azure.Identity/src/MsalPublicClient.cs

```diff
@@ -35,34 +40,74 @@ public MsalPublicClient(HttpPipeline pipeline, string clientId, string tenantId
 
             _client = pubAppBuilder.Build();
 
-            if (attachSharedCache)
+            _clientId = clientId;
+
+            _ensureInitAsync = new Lazy<Task>(InitializeAsync);
```

Comment:
If it matters, `LazyInitializer.EnsureInitialized` using a factory is cheaper than a `Lazy<T>`.